### PR TITLE
Added unit tests for Kubernetes module

### PR DESCRIPTION
**Pull request by brejoc**
### What does this PR do?

Added unit tests for:
    * Node listing,
    * deployment listing,
    * service listing,
    * pod listing,
    * deployment deletion and
    * deployment creation.

### What issues does this PR fix or reference?

None

### Tests written?

Yes ;)

### CODE REVIEW

File: tests/unit/modules/kubernetes_test.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+'''
+    :codeauthor: :email:`Jochen Breuer <jbreuer@suse.de>`
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+import os
+
+# Import Salt Testing Libs
+from salttesting import TestCase, skipIf
```
Comment (rallytime) [inline tests/unit/modules/kubernetes_test.py]: Can you update this import to `from tests.support.unit import TestCase, skipIf`

File: tests/unit/modules/kubernetes_test.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+from salttesting.mock import (
```
Comment (rallytime) [inline tests/unit/modules/kubernetes_test.py]: This import also needs to be updated to `from tests.support.mock import (`

We moved a bunch of the salttesting libs around for 2017.7 and onward.

File: tests/unit/modules/kubernetes_test.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+'''
+    :codeauthor: :email:`Jochen Breuer <jbreuer@suse.de>`
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+import os
```
Comment (rallytime) [inline tests/unit/modules/kubernetes_test.py]: This import isn't used.

**Review by rallytime — CHANGES_REQUESTED** [review]
Thanks for adding tests @brejoc! We always love that. :)

For the 2017.7 branch, we moved some things around in salttesting so we can use pytest. Can you rename this file `test_kubernetes.py` instead of kubernets_test? I also added a couple of comments that need to be fixed for lint.

Once the file is renamed and the other lint issues are fixed, we can see how the test run goes.

**Comment by brejoc** [comment]
@rallytime The Kubernetes moduels has the [kubernetes](https://pypi.python.org/pypi/kubernetes/) client lib as a dependency. How can we make this happen for the test?

```14:27:16 ImportError: No module named kubernetes.client```

**Comment by cachedout** [comment]
@brejoc There are some lint errors here. Let us know if you need any help resolving them.

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/13863/violations/file/tests/unit/modules/test_kubernetes.py/

**Comment by brejoc** [comment]
@cachedout Sorry to bother you, but are the results for the linting job already gone? The link shows a 404.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Hi @brejoc - I can add the kubernetes lib to our jenkins test prep states. In the mean time, these tests shouldn't be failing during a test-run out of the box. Can you gate these tests to skip if the required libs are not present?

**Comment by brejoc** [comment]
@rallytime Okay, I think this should be fine now. What do you think?

**Comment by rallytime** [comment]
@brejoc I think that looks good. Thank you!

I have added a state to install the kubernetes python client to our test VMs and I also updated your branch to the latest HEAD of 2017.7 so we can get a clean run and make sure your tests run accordingly.

**Review by rallytime — APPROVED** [review]
Pending latest test run, this looks good!

**Pull request merged**